File: LeetCode-Problems/Easy/Range-Sum-BST.py
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        q = list()
        q.append(root)
        s = 0
        while (len(q) != 0):
            curr = q.pop(0)
            if (curr.val >= low and curr.val <= high):
                s = s + curr.val
            if (curr.left != None):
                q.append(curr.left)
            if (curr.right != None):
                q.append(curr.right)
        return s

File: LeetCode-Problems/Easy/Range-Sum-BST.java
class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
        Queue<TreeNode> q = new LinkedList<TreeNode>();
        q.add(root);
        int sum = 0;
        while (!q.isEmpty()){
            TreeNode curr = q.poll();
            if (curr.val >= low && curr.val <= high){
                sum += curr.val;
            }
            if (curr.left != null){
                q.add(curr.left);
            }
            if (curr.right != null){
                q.add(curr.right);
            }
        }
        return sum;
    }
}
